astronomer/providers/http/example_dags/example_external_deployment_task_sensor.py

```diff
@@ -0,0 +1,78 @@
+import os
+from datetime import timedelta
+
+import requests
+from airflow import DAG
+from airflow.hooks.base import BaseHook
+from airflow.operators.python import PythonOperator
+from airflow.utils.log.secrets_masker import mask_secret
+from airflow.utils.timezone import datetime
+
+from astronomer.providers.http.sensors.external_deployment_task import (
+    ExternalDeploymentTaskSensorAsync,
+)
+
+DEPLOYMENT_CONN_ID = os.getenv("ASTRO_DEPLOYMENT_CONN_ID", "deployment_conn_id")
+ASTRONOMER_KEY_ID = os.getenv("ASTRONOMER_KEY_ID", "")
+ASTRONOMER_KEY_SECRET = os.getenv("ASTRONOMER_KEY_SECRET", "")
+EXECUTION_TIMEOUT = int(os.getenv("EXECUTION_TIMEOUT", 6))
+DAG_ID = os.getenv("DAG_ID", "")
+RUN_ID = os.getenv("RUN_ID", "")
+TASK_ID = os.getenv("TASK_ID", "")
+
+default_args = {
+    "execution_timeout": timedelta(hours=EXECUTION_TIMEOUT),
+    "retries": int(os.getenv("DEFAULT_TASK_RETRIES", 2)),
+    "retry_delay": timedelta(seconds=int(os.getenv("DEFAULT_RETRY_DELAY_SECONDS", 60))),
+}
+HEADERS = {
+    "cache-control": "no-cache",
+    "content-type": "application/json",
+    "accept": "application/json",
+}
+
+
+def astro_access_token() -> None:
+    """Get the access_token by making post request with client_id"""
+    conn = BaseHook.get_connection(DEPLOYMENT_CONN_ID)
+    _json = {
+        "audience": "astronomer-ee",
+        "grant_type": "client_credentials",
+        "client_id": ASTRONOMER_KEY_ID,
+        "client_secret": ASTRONOMER_KEY_SECRET,
+    }
+    token_resp = requests.post(
+        url=conn.host,
+        headers={"Content-type": "application/json"},
+        json=_json,
+    )
+    masked_access_token = token_resp.json()["access_token"]
+    mask_secret(masked_access_token)
+    return HEADERS.update({"Authorization": "Bearer " + masked_access_token})
```

Comment:
```suggestion


def astro_access_token() -> None:
    """Get the access_token by making post request with client_id"""
    conn = BaseHook.get_connection(DEPLOYMENT_CONN_ID)
    _json = {
        "audience": "astronomer-ee",
        "grant_type": "client_credentials",
        "client_id": ASTRONOMER_KEY_ID,
        "client_secret": ASTRONOMER_KEY_SECRET,
    }
    token_resp = requests.post(
        url=conn.host,
        headers={"Content-type": "application/json"},
        json=_json,
    )
    masked_access_token = token_resp.json()["access_token"]
    mask_secret(masked_access_token)
    return {
        "cache-control": "no-cache",
        "content-type": "application/json",
        "accept": "application/json",
        "Authorization": "Bearer " + masked_access_token
    }
```

If you accept the above, please update the function name as it generates the entire header now.

Reply:
Covered it now